Product Gallery: hide next/prev buttons on mobile

kmanijak opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

We apply opinionated mobile layout in Product Gallery. Part of it is hiding Next/Prev buttons and rely on Thumbnails and swiping gestures for navigation.

Closes WOOPLUG-3794

(For Bug Fixes) Bug introduced in PR # .

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

Case | Before | After
 ------ | ----- | --
 Editor   | <img width="774" alt="image" src="https://github.com/user-attachments/assets/29850697-3914-43d3-bbad-c5f4abb3d2ef" />    |    <img width="872" alt="image" src="https://github.com/user-attachments/assets/422c22db-11e0-4a1b-9fd5-ccde1879affc" />   |
Desktop | <img width="403" alt="image" src="https://github.com/user-attachments/assets/cc52e08d-1f9d-4f21-98ac-a210d80cbd91" /> | <img width="407" alt="image" src="https://github.com/user-attachments/assets/97e9e6ac-bf92-4808-b7b6-3276c8b2f027" />


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Single Product template
2. Replace Product Image Gallery with Product Gallery (Beta)
3. **Expected:** Arrows are there
4. Change the Editor view to mobile
5. **Expected:** Arrows disappear in mobile version
6. Save and go to frontend
7. **Expected:** Arrows are there
8. Enter mobile view or decrease the viewport width below 600px
9. **Expected:** Arrows disappear in mobile version

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

kmanijak committed:
Simplify the SCSS

kmanijak committed:
Hide Next/prev buttons on mobile

kmanijak committed:
Add test

kmanijak committed:
Add changelog

dinhtungdu reviewed (approved):
Thanks for working on the PR! The PR is looking and testing good for me. :shipit: 

#### 🖼️ Screenshots/ screen recordings

<img width="1639" alt="image" src="https://github.com/user-attachments/assets/3434203a-7eac-466c-94d0-762902a7bb5b" />

#### 📝 List of functionality tested / steps followed

Tested the mobile view both in the editor and on the front end.

#### 🌐 Site details
<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.9.0-dev
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.2
Log Directory Writable: ✔
WP Version: ❌ 6.7.2 - There is a newer version of WordPress available (6.8)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.25.4
Server Architecture: Darwin 24.4.0 arm64
PHP Version: 7.4.33
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 8.5.0
OpenSSL/3.1.4

SUHOSIN Installed: –
MySQL Version: 8.0.33
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 23
global_product_addon: 1
page: 8
post: 2
product: 18
product_variation: 7
revision: 7
wp_font_face: 36
wp_font_family: 12
wp_global_styles: 1
wp_navigation: 1
wp_template: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 9.9.0-dev

### Inactive Plugins (6) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.6
Hello Dolly: by Matt Mullenweg – 1.7.2
Laravel DD for Wordpress: by Peter Hegman – 1.0.1
WooCommerce: by Automattic – 9.7.1 (update to version 9.8.1 is available)
WooCommerce Beta Tester: by WooCommerce – 3.0.0
WooCommerce Product Add-Ons: by Woo – 7.8.1

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: HKD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –
Enabled Features: analytics
marketplace
order_attribution
site_visibility_badge
remote_logging
email_improvements
custom_order_tables


### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 48 KB

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/ -  Contains the woocommerce/cart block
Checkout: #9 - /checkout/ -  Contains the woocommerce/checkout block
My account: #10 - /my-account/ -  Contains the [woocommerce_my_account] shortcode
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Five
Version: 1.0 (update to version 1.2 is available)
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Block theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /Users/tung/workspace/woocommerce/plugins/woocommerce/templates/block-notices/error.php
/Users/tung/workspace/woocommerce/plugins/woocommerce/templates/block-notices/notice.php
/Users/tung/workspace/woocommerce/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
experimental-blocks
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
minified-js
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
payment-gateway-suggestions
product-custom-fields
printful
remote-inbox-notifications
remote-free-extensions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
async-product-editor-category-field
launch-your-store
add-to-cart-with-options-stepper-layout
blockified-add-to-cart
disable-core-profiler-fallback
reactify-classic-payments-settings

Disabled Features: product-data-views
coming-soon-newsletter-template
product-pre-publish-modal
settings
product-editor-template-system
use-wp-horizon

Daily Cron: ✔ Next scheduled: 2025-04-16 08:18:38 +00:00
Options: ✔
Notes: 63
Onboarding: completed

### Action Scheduler ###

Complete: 162
Oldest: 2025-03-17 08:19:43 +0000
Newest: 2025-04-16 09:33:41 +0000

Failed: 1
Oldest: 2025-03-17 08:19:52 +0000
Newest: 2025-03-17 08:19:52 +0000

Pending: 1
Oldest: 2025-04-17 09:33:41 +0000
Newest: 2025-04-17 09:33:41 +0000


### Status report information ###

Generated at: 2025-04-16 09:33:44 +00:00
```
</details>

kmanijak committed:
Merge branch 'trunk' into wooplug-3794-product-gallery-hide-nextprev-buttons-on-mobile

triple0t committed:
Add $email_object parameter to woocommerce_email_preview_placeholders filter (#57272)

* Add $email_object parameter to woocommerce_email_preview_placeholders filter for enhanced email preview functionality

* Update docblocks for get_placeholders method to clarify $email_object parameter type

Technically, WC Core emails only use WC_Order or WP_User, but an extension can register email with any object, in which case, the type hint can be misleading.

* Add @since for a newly added parameter

---------

Co-authored-by:  Ján Mikláš <neosinner@gmail.com>

kmanijak committed:
Fix E2E tests after WP 6.8 release (#57320)

* Fix Accordion E2E tests after WP 6.8 release

* Add changelog

* Fix Checkout -> Attributes -> Shipping and billing addresses › Apartment input visibility and optional and required can be toggled

mikejolley committed:
[Checkout] Local Pickup rates should be available regardless of shipping restrictions (#56922)

* Update error strings for consistency

* Calculate local pickup rates for all locations

* Utils to assist with packages

* Skip country validation if collectng

* Missing local pickup ids in order class

* Widen countries display in checkout to include possible billing and shipping countries

* changelog

* Revert text change to reduce diff

* Revert tweak to reduce diff size

* Reduce diff size by using old structure

* Change validation order to reduce diff size

* Remove extra escaping to reduce diff size

* Fix indenting

* Add helper for array_all

* Use array_all helper

* Add test coverage for flows involving local pickup

* Validate allowed countries in client

* Unused variable

* undo text change to reduce diff

* Leave restrictions in place for classic checkout

* Move util

* Update tests to use pickup location shipping method

* validate_addresses escaping

* Missing docblock

gigitux committed:
Enhance IP address retrieval in WC_Geolocation to handle multiple IPs in REMOTE_ADDR (#57284)

* Enhance IP address retrieval in WC_Geolocation to handle multiple IPs in REMOTE_ADDR. Update unit tests to verify correct IP extraction from comma-separated values.

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

mikejolley committed:
[Additional Fields] Additional Information Step should be hidden if empty (#57041)

* Fix eslint warnings in IDE

* Export useFormFields

* Update Form so its passed fields from the parent

* TS updates

* Simplify address wrapper usage

* Update AddressWrapper usage

* Prevent interaction in editor with additional fields

* Hide addiitonal fields if fields are empty

* Undo form changes

* changelog

kmanijak committed:
Fix flaky shipping and billing e2e tests (#57326)

* Add expect().toPass() to checking and unchecking radio buttons as it's flaky

* Add changelog

gigitux committed:
registerProductBlockType: Re-enable the test for block visibility after addressing flakiness (#57325)

* registerProductBlockType: Re-enable the test for block visibility after addressing flakiness

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

gigitux committed:
Add blueprint CLI initialization to 'init' hook to prevent premature translation loading in WC_CLI class (#57291)

* Add blueprint CLI initialization to 'init' hook to prevent premature translation loading in WC_CLI class

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

AhmarZaidi committed:
fix: Add loaded state to hand picked control (#57300)

Co-authored-by: Seghir Nadir <nadir.seghir@gmail.com>
Co-authored-by: Luigi Teschio <gigitux@gmail.com>

martin-neumann-gurus committed:
Allow Password Reset for Users that are not members of the current blog of Multisite Network (#57242)

jorgeatorres committed:
Prevent possible fatal in 9.9 update routine (#57257)

* Prevent possible fatal in 9.9 update routine
* Add changelog

amitraj2203 committed:
Update Product Stock Indicator block description to explain empty state behavior (#57304)

* feat: Update description for Product Stock Indicator block to clarify functionality

* feat: Add changelog file

kmanijak committed:
Product Gallery: improve UI for Next/Prev buttons (#57135)

* Enable supports in arrows

* Provide basic supports

* Simplify the arrows structure

* Align the frontend with editor

* Add some basic styling

* Fix lint

* Fix lint

* Update icons and align them correctly

* Improve classnames and styling

* Apply changes on the frontend

* Add supports styles in editor

* Support colors in editor

* Simplify the arrows structure even more

* Change sprintf to ob_clean

* Fix styles

* Add support for shadow

* Add shadow support in StyleAttributesUtils

* Fix lint

* Change the disabled arrow opacity from 20% to 35%

* Add changelog

* Update selectors in tests

* Remove unnecessary z-index

* Fix the editor util for getting next/prev button

* Replace str_contains with strpos for PHP 7 support
